platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponentActivationOptions.java

```diff
@@ -0,0 +1,184 @@
+   * @return
+   */
+  @Nullable
+  public LocationEngineRequest locationEngineRequest() {
+    return locationEngineRequest;
+  }
+
+  /**
+   * A built {@link LocationComponentOptions} object, which holds the various {@link LocationComponent} styling options
+   * @return the options for styling the actual LocationComponent
+   */
+  @Nullable
+  public LocationComponentOptions locationComponentOptions() {
+    return locationComponentOptions;
+  }
+
+  /**
+   * the LocationComponent style resource (e.g. R.style.style_name)
+   *
+   * @return the style resource.
+   */
+  @Nullable
+  public int styleRes() {
+    return styleRes;
+  }
+
+  /**
+   * True if you want to initialize and use the built-in location engine or false if there should be no
+   * location engine initialized
+   * @return whether the default LocationEngine is used
+   */
+  @Nullable
+  public boolean useDefaultLocationEngine() {
+    return useDefaultLocationEngine;
+  }
+
+  /**
+   * Builder class for constructing a new instance of {@link LocationComponentActivationOptions}.
+   *
+   * since 7.2.0
+   */
+  public static class Builder {
+    private Context context;
+    private Style style;
+    private LocationEngine locationEngine;
+    private LocationEngineRequest locationEngineRequest;
+    private LocationComponentOptions locationComponentOptions;
+    private int styleRes;
+    private boolean useDefaultLocationEngine;
+
+    public Builder(Context context, Style style) {
+      this.context = context;
+      this.style = style;
+    }
+
+    public Builder locationEngine(LocationEngine locationEngine) {
+      this.locationEngine = locationEngine;
+      return this;
+    }
+
+    public Builder locationEngineRequest(LocationEngineRequest locationEngineRequest) {
+      this.locationEngineRequest = locationEngineRequest;
+      return this;
+    }
+
+    public Builder locationComponentOptions(LocationComponentOptions locationComponentOptions) {
+      this.locationComponentOptions = locationComponentOptions;
+      return this;
+    }
+
+    public Builder styleRes(int styleRes) {
+      this.styleRes = styleRes;
+      return this;
+    }
+
+    public Builder useDefaultLocationEngine(boolean useDefaultLocationEngine) {
+      this.useDefaultLocationEngine = useDefaultLocationEngine;
+      return this;
+    }
+
+    public LocationComponentActivationOptions build() {
+      if (styleRes != 0 && locationComponentOptions != null) {
+        throw new IllegalArgumentException(
+          "You've provided both a style resource and a LocationComponentOptions object to the " +
+            "LocationComponentActivationOptions builder. You can't use both and " +
+            "you must choose one of the two to style the LocationComponent.");
+      }
+      if (context == null) {
+        throw new NullPointerException(
+          "Context in LocationComponentActivationOptions is null.");
+      }
+      if (style == null) {
```

Comment:
You should also check if the style is fully loaded.

Reply:
Ok, I now have:

```java
if (style == null) {
        throw new NullPointerException(
"Style in LocationComponentActivationOptions is null. Make sure the Style object isn't null." +
            " Wait for the map to fully load before passing the Style object to LocationComponentActivationOptions.");
}

if (!style.isFullyLoaded()) {
        throw new IllegalArgumentException(
          "Style in LocationComponentActivationOptions isn't fully loaded. Wait for the map to fully load before " +
            "passing the Style object to LocationComponentActivationOptions.");
}
```